services/prometheus/prometheus.go

```diff
@@ -27,8 +30,26 @@ import (
 	"github.com/percona/pmm-managed/models"
 )
 
-func AddScrapeConfigs(l *logrus.Entry, cfg *config.Config, q *reform.Querier, s *models.MetricsResolutions) error {
-	agents, err := q.SelectAllFrom(models.AgentTable, "WHERE NOT disabled AND listen_port IS NOT NULL ORDER BY agent_type, agent_id")
+// AddScrapeConfigs - adds agents scrape configuration to given scrape config,
+// agent filter can be used for agents filtering.
+func AddScrapeConfigs(l *logrus.Entry, cfg *config.Config, q *reform.Querier, s *models.MetricsResolutions, filter models.AgentFilters) error {
+	var (
+		args       []interface{}
+		conditions []string
+	)
+	if filter.PMMAgentID != "" {
```

Comment:
can we move this part to models package?

Reply:
where clause generation?

Reply:
Yes